weatherman.py

```diff
@@ -0,0 +1,213 @@
+import fnmatch
+import os
+import sys
+import calendar
+
+highest_temp= ['date',-50]                     # highest_temp[0] would contain date and temp[1] would contain max_temp on that day
+lowest_temp= ['date',50]
+highest_humidity=['date',-50]
+
+
+# def user_input():
+
+#     year=input()
+#     return year
+
+
+
+
+def file_name_matching(year,month_name):                         #send null string in case of ingle argument
+
+     for file_name in os.listdir('.'):
+
+        if month_name=="":
+            if fnmatch.fnmatch( file_name, '*'+year+'*'):        # check wether file name matches with pattern given as second argument. '*' means could be anything
+                file_parsing_year(file_name)
+
+        elif fnmatch.fnmatch(file_name, '*'+year+'_'+month_name[:3]+'*'):  #[:3] because file name only consist 3 alphabets of month_name
```

Comment:
Slicing with a hardcoded index is never a good approach. If format gets changed, or name is incorrect and you can't get something on `3` what will happen? We need to think of such edge cases while writing code.